boa/src/object/jsarray.rs

```diff
@@ -0,0 +1,344 @@
+use std::ops::Deref;
+
+use crate::{
+    builtins::Array,
+    gc::{Finalize, Trace},
+    object::JsObject,
+    Context, JsResult, JsString, JsValue,
+};
+
+// TODO: put in a better place.
+pub trait JsObjectType: Into<JsValue> + Into<JsObject> + Deref<Target = JsObject> {}
+
+/// JavaScript `Array` rust object.
+#[derive(Debug, Clone, Trace, Finalize)]
+pub struct JsArray {
+    inner: JsObject,
+}
+
+impl JsArray {
+    #[inline]
+    pub fn new(array: &[JsValue], context: &mut Context) -> Self {
```

Comment:
We could copy the API of `std::Vec`, making `new` create a new, empty `JsArray`, `From` implementations for `&[JsValue]` and friends, `TryFrom<JsObject>` and a `FromIterator` implementation that uses `create_array_from_list`

Reply:
> We could copy the API of `std::Vec`, making `new` create a new, empty `JsArray`, `from` methods for `&[JsValue]` and friends, `try_from::<JsObject>` and a `from_iter` that uses `create_array_from_list`

Sadly, the `From`, `TryFrom`, and `FromIterator` cannot be implemented since they only allow one argument, but we also need the `context`.

Reply:
Maybe as associated functions?